infra.bs

```diff
@@ -258,11 +262,36 @@ inclusive.
 
 <h3 id=strings>Strings</h3>
 
-<p>A <dfn export>string</dfn> is a sequence of <a>code points</a>. Strings are denoted by double
-quotes and monospace font.
+<p>A <dfn export>JavaScript string</dfn> is a sequence of unsigned 16-bit integers, also known as
+<dfn export lt="code unit">code units</dfn>. A <a>JavaScript string</a> can also be interpreted as
+containing <a>code points</a>, per the conversion defined in <a>The String Type</a> section of the
+JavaScript specification. [[!ECMA-262]]
+
+<p class=note>This is different from how the Unicode specification defines "code unit". In
+particular it refers exclusively to an unsigned 16-bit integer as is similar to how "code unit" is
+defined for UTF-16. However, therefore it can also refer to a surrogate that is not part of a
+surrogate pair, which is not the case for "code unit" as defined for UTF-16. [[UNICODE]]
+
+<p>A <dfn export>scalar value string</dfn> is a sequence of <a>scalar values</a>.
+
+<p class=note>A <a>scalar value string</a> is useful for any kind of I/O or other kind of operation
+where <a>UTF-8 encode</a> comes into play.
+<!-- It's also useful if you can imagine the subsystem to be implemented in Rust -->
+
+<p><dfn export lt=string>String</dfn> can be used to refer to either a <a>JavaScript string</a> or
+<a>scalar value string</a>, when it is clear from the context which is meant or when the distinction
+is immaterial. <a>Strings</a> are denoted by double quotes and monospace font.
 
 <p class=example id=example-string-notation>"<code>Hello, world!</code>" is a string.
 
+<p>To <dfn export for="JavaScript string">convert</dfn> a <a>JavaScript string</a> into a
+<a>scalar value string</a>, replace any <a>surrogate code points</a> with U+FFFD.
```

Comment:
I would probably add the word "unpaired" outside the anchor tag for clarity. I know that this phrasing is correct, but you never quite say that properly paired surrogate code units make a supplementary code point. I think it might be useful to say that this conversion results in a (well-formed) UTF-16 string.